getter.go

```diff
@@ -119,7 +154,60 @@ func (g *getter) retryRequest(method, urlStr string, body io.ReadSeeker) (resp *
 	return
 }
 
-func (g *getter) initChunks() {
+func (bg *batchGetter) queueFile(path string) error {
```

Comment:
This had me thinking that really the purpose of the works is to download chunks.  It doesn't really care whose chunk it is downloading, just a bunch of chunk downloaders.  It matters when you are stitching results back together, in our single getter case, we want all the chunks in a single file / byte buffer;  In our multi getter we want the end results in a single file that is concat in order.   But you could break that multi getter problem into concurrent single getters that just know their position to write to.   We will need to sit down to think it through some more, but I feel we can have a solution that implements the WriteTo interface and the single getter knows where it is writing to and can accept an offset for the write, and the chunk writing would use that offset plus the chunk offset.